### PR TITLE
Don't pass std::strings across shared libs in appleseed.python

### PR DESCRIPTION
**Pull request by est77**

### CODE REVIEW

File: src/appleseed.python/murmurhashwrapper.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#include "foundation/platform/python.h"
+#include "foundation/utility/murmurhash.h"
+
+// Standard headers.
+#include <string>
+
+class MurmurHashWrapper
+{
+  public:
+
```
Comment (dictoon) [inline src/appleseed.python/murmurhashwrapper.h]: Remove blank line.

File: src/appleseed.python/murmurhashwrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{
+    // Null terminator character.
+    m_string[CharBufferSize-1] = 0;
+}
+
+const char* MurmurHashWrapper::to_string()
+{
+    const std::string str = m_hash.to_string();
+
+    memcpy(m_string, str.c_str(), CharBufferSize - 1);
```
Comment (dictoon) [inline src/appleseed.python/murmurhashwrapper.cpp]: You need to prefix this with `std::` and include `<cstring>`.

File: src/appleseed.python/murmurhashwrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// appleseed.foundation headers.
+#include "foundation/utility/murmurhash.h"
+
+// Standard headers.
+#include <string>
+
+MurmurHashWrapper::MurmurHashWrapper()
+{
+    // Null terminator character.
+    m_string[CharBufferSize-1] = 0;
```
Comment (dictoon) [inline src/appleseed.python/murmurhashwrapper.cpp]: Add spaces around `-`.

File: src/appleseed.python/CMakeLists.txt [hunk 4]
```diff
@@ -100,6 +100,8 @@ set (sources
     logtarget.py
     metadata.h
     module.cpp
+    murmurhashwrapper.h
```
Comment (dictoon) [inline src/appleseed.python/CMakeLists.txt]: Order alphabetically (.cpp before .h).

File: src/appleseed.python/murmurhashwrapper.cpp [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "murmurhashwrapper.h"
+
+// appleseed.foundation headers.
+#include "foundation/utility/murmurhash.h"
```
Comment (dictoon) [inline src/appleseed.python/murmurhashwrapper.cpp]: Redundant, already included in the header.

File: src/appleseed.python/murmurhashwrapper.h [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    bool operator<(const MurmurHashWrapper& other) const
+    {
+        return m_hash < other.m_hash;
+    }
+
+    const char* to_string();
+
+    enum {CharBufferSize = 2 * sizeof(foundation::MurmurHash) + 1};
```
Comment (dictoon) [inline src/appleseed.python/murmurhashwrapper.h]: Add spaces around braces.

File: src/appleseed.studio/CMakeLists.txt [hunk 7]
```diff
@@ -264,6 +264,8 @@ set (appleseedpython_sources
     ../appleseed.python/logtarget.py
     ../appleseed.python/metadata.h
     ../appleseed.python/module.cpp
+    ../appleseed.python/murmurhashwrapper.h
```
Comment (dictoon) [inline src/appleseed.studio/CMakeLists.txt]: Order alphabetically.

**Comment by est77** [comment]
Replaced by #2076

**Pull request closed without merging**